visionSamples/barcode-reader/app/src/main/java/com/google/android/gms/samples/vision/barcodereader/BarcodeGraphicTracker.java

```diff
@@ -26,25 +29,43 @@
  * to an overlay, update the graphics as the item changes, and remove the graphics when the item
  * goes away.
  */
-class BarcodeGraphicTracker extends Tracker<Barcode> {
+public class BarcodeGraphicTracker extends Tracker<Barcode> {
     private GraphicOverlay<BarcodeGraphic> mOverlay;
     private BarcodeGraphic mGraphic;
 
-    BarcodeGraphicTracker(GraphicOverlay<BarcodeGraphic> overlay, BarcodeGraphic graphic) {
-        mOverlay = overlay;
-        mGraphic = graphic;
+    public UpdateBarcodeListener updateBarcodeListener;
+
+    public interface UpdateBarcodeListener {
+        @UiThread
+        void getBarcode(Barcode barcode);
+    }
+
+    BarcodeGraphicTracker(GraphicOverlay<BarcodeGraphic> mOverlay, BarcodeGraphic mGraphic,
+                          Context context) {
+        this.mOverlay = mOverlay;
+        this.mGraphic = mGraphic;
+
+        if (context instanceof UpdateBarcodeListener) {
+            updateBarcodeListener = (UpdateBarcodeListener) context;
```

Comment:
this.mUpdateBarcodeListener = (UpdateBarcodeListener) context;

Reply:
Done.